src/moscot/problems/base/_utils.py

```diff
@@ -228,3 +236,242 @@ def _filter_kwargs(*funcs: Callable[..., Any], **kwargs: Any) -> Dict[str, Any]:
         params = inspect.signature(func).parameters
         res.update({k: v for k, v in kwargs.items() if k in params})
     return res
+
+
+@d.dedent
+def _correlation_test(
+    X: Union[ArrayLike, spmatrix],
+    Y: pd.DataFrame,
+    feature_names: Sequence[str],
+    method: CorrTestMethod = CorrTestMethod.FISCHER,
+    confidence_level: float = 0.95,
+    n_perms: Optional[int] = None,
+    seed: Optional[int] = None,
+    **kwargs: Any,
+) -> pd.DataFrame:
+    """
+    Perform a statistical test for correlation between X and .
+
+    Return NaN for genes which don't vary across cells.
+
+    Parameters
+    ----------
+    X
+        Array or sparse matrix of shape ``(n_cells, n_features)`` containing the counts.
+    Y
+        Data frame of shape ``(n_cells, 1)`` containing the pull/push distribution.
```

Comment:
Afaik in the `CellRank` code, this can be `(n_cells, n_lineages)`, why only 1? It will have worse performance.